samcli/lib/build/app_builder.py

```diff
@@ -359,6 +358,14 @@ def _build_lambda_image(self, function_name: str, metadata: Dict, architecture:
         docker_build_target = metadata.get("DockerBuildTarget", None)
         docker_build_args = metadata.get("DockerBuildArgs", {})
 
+        if not dockerfile or not docker_context:
+            LOG.info(
+                "Skip Building %s function, as it does not contain either Dockerfile or DockerContext "
+                "metadata properties.",
+                function_name,
+            )
+            return ""
```

Comment:
nit: should we return `None` or empty string?

Reply:
should it raise an exception here instead?

Reply:
@mndeveci .. I returned empty string as the method was defined to return str not Optional[str] .. but I believe I can make it None

Reply:
@torresxb1 .. I think we should not raise an exception, the function can be referring to a remote or local image uri, and so no need to include the  image build related metadata, so sam will skip building it.